clients/venice-client/src/main/java/com/linkedin/venice/fastclient/DispatchingAvroGenericStoreClient.java

```diff
@@ -572,6 +571,9 @@ private void batchGetTransportRequestCompletionHandler(
     VeniceCompressor compressor =
         metadata.getCompressor(transportClientResponse.getCompressionStrategy(), requestContext.currentVersion);
     for (MultiGetResponseRecordV1 r: records) {
+      // Since the underlying FastSerializerDeserializerFactory is caching deserializers by reader, writer and object
+      // class. Retrieving the corresponding deserializer per record should be fast here after initial warmup.
+      RecordDeserializer<V> dataRecordDeserializer = getDataRecordDeserializer(r.getSchemaId());
```

Comment:
Nitpick: I think we could move this line to just after `nanoTsBeforeDeserialization`, since looking up the deserializer is technically part of the deserialization time.

Also, another nitpick in that area... we do:

```
      totalDecompressionTimeForResponse += System.nanoTime() - nanoTsBeforeDecompression;
      long nanoTsBeforeDeserialization = System.nanoTime();
```

I don't think we need to do two `System.nanoTime()` calls in a row, we should be able to do only one, e.g. something like:

```
      long nanoTsBeforeDeserialization = System.nanoTime();
      V deserializedValue = dataRecordDeserializer.deserialize(decompressRecord);
      requestContext.recordRecordDeserializationTime(
          transportClientResponse.getRouteId(),
          getLatencyInNS(nanoTsBeforeDeserialization));
      MultiKeyRequestContext.KeyInfo<K> k = keyInfos.get(r.keyIndex);
      keysSeen.add(r.keyIndex);
      callback.onRecordReceived(k.getKey(), deserializedValue);
      totalDecompressionTimeForResponse += nanoTsBeforeDeserialization - nanoTsBeforeDecompression;
```

And one more nitpick here (last one, I promise 😅 ...), why do we measure the per-record deserialization time and the total decompression time? Why not measure the total time in both cases? If we're trying to analyze the client's performance and figure out where the bottleneck is, it seems confusing and counter-productive to have these two measurements at different granularity, because then we would need to normalize by the number of keys (per route!) in order to compare them to one another. Finding out what is the bigger bottleneck would be non-straightforward...

Reply:
Great suggestions, will fix.

Reply:
Addressed the first two points. I think Nisarg already fixed the third point in his refactor since recordRecordDeserilizationTime is aggregating by routeID
```
  void recordRecordDeserializationTime(String routeId, long latencyInNS) {
    Validate.notNull(routeId);
    routeRequests.get(routeId).recordDeserializationTime.addAndGet(latencyInNS);
  }
```

Reply:
Sure. It seems a bit unfortunate to do the `routeRequests.get(routeId)` lookup per record but that's ok... No need to bloat the current PR. Thanks for addressing the first two!